Add market closure governance proposal (#1614)
vegaprotocol/specs: protocol/0028-GOVE-governance.md

Not in scope: minimum participation of active users, i.e. 90% of the _active_ users of the vega network have to take part in the vote. Minimum participation is currently always measured against the total possible participation.

### Market change proposal outcome

For market change proposals the network will additionally calculate

1. `LP participation rate = SUM (equity-like share of all LPs who cast a vote)` (no need to divide by anything as equity-like share sums up to `1`).

> 1. `LP for rate = SUM (equity-like share of all LPs who cast a for vote)) / LP participation rate`.

Reviewer: ```suggestion
1. `LP for rate = SUM (equity-like share of all LPs who cast a for vote))`.
```

## 5. Transfers initiated by Governance (post Oregon trail)

## 6. Close market

> Any type of market (either fixed expiry market or perpetual) can be closed via a governance vote.

Reviewer: We should make clear that it's the same voting rules as market update proposals (i.e. LPs vote via ELS, can be overriden by governance token holders).

Author: I remember that being mentioned last time we discussed it, but couldn't find anything in the specs about it (probably was looking in a wrong place)

A proposal to close a market contains:

> 1. final settlement price

Reviewer: I wonder how to handle this for a spot market (which we want to allow to be closed) but it won't need a final settlement price.

Author: We can either build more complex validation where market validates proposal that's meant to apply to it or just always require some price, but spot would ignore it (not elegant, but easiest to implement)

4. Enacted governance transfers are therefore available to be queried via the regular transfer API in data node.
5. Governance initiated transfers are subject to neither minimum transfer amounts nor to fees.

## 6. Change market state

A governance proposal to change the state of the market.
Multiple concurrent proposals are allowed. If more than one gets successfully voted in the one which arrived last gets used. If a proposal has already been successful then additional market state change proposals are still allowed as long as their enactment date is no later then that of the last successful vote.

Refer to subsections below for allowed state changes.

### 6.1. Move market to a closed state

Any type of market (either fixed expiry market or perpetual) can be closed via a governance vote.

A proposal to close a market contains:

1. an enactment time
1. final settlement price (not required for spot markets)

> Once market is closed the process cannot be reversed.

Reviewer: Do the market change proposal voting rules apply? I would suggest they do. 

Btw. on that we have that only an LP with a certian ELS can make the proposal (min tokens *and* min ELS met). I think we should change those to (min tokens *OR* min ELS met).

Author: Clarified now, happy to add the modification you mentioned, but perhaps in a separate PR?

Reviewer: Let's do separate PR as it will be cleaner to track implementation.

- A market change proposal that's to modify any parameters on a market in `pending` state (i.e. voting has successfully completed and the market is in the opening auction) will be accepted and if it's the enactment time happens to be before the opening auction ends then the proposed modification is enacted. (<a name="0028-GOVE-070" href="#0028-GOVE-070">0028-GOVE-070</a>)
- In particular a market change proposal that's to modify the parent market on a market in `pending` state (i.e. voting has successfully completed and the market is in the opening auction) will be accepted and if it's the enactment time happens to be before the opening auction ends then the parent is used (assuming the proposed parent doesn't already have a successor). (<a name="0028-GOVE-071" href="#0028-GOVE-071">0028-GOVE-071</a>)
- A market change that's to modify any parameters on a market in `pending` state (i.e. voting has successfully completed on the market creation and the market is in the opening auction) will run voting rules the same as market creation proposals i.e. LPs don't get a vote. (<a name="0028-GOVE-072" href="#0028-GOVE-072">0028-GOVE-072</a>)

> - A governance proposal to close a market which doesn't specify the final settlement price gets rejected buy the markets which require it (non-spot). (<a name="0028-GOVE-108" href="#0028-GOVE-108">0028-GOVE-108</a>)

Reviewer: ```suggestion
- A governance proposal to close a market which doesn't specify the final settlement price gets rejected by the markets which require it (non-spot). (<a name="0028-GOVE-108" href="#0028-GOVE-108">0028-GOVE-108</a>)
```

> - When multiple market closure governance proposals are submitted and accepted then one to arrive last is the one whose enactment time and final settlement price get used to close the market (<a name="0028-GOVE-109" href="#0028-GOVE-109">0028-GOVE-109</a>)

Reviewer: ```suggestion
- When multiple market closure governance proposals are submitted and accepted then one to be approved by governance last (i.e latest closing date and vote goes through) is the one whose enactment time and final settlement price get used to close the market (<a name="0028-GOVE-109" href="#0028-GOVE-109">0028-GOVE-109</a>)
```
